### PR TITLE
Remove All from all the methods #319

### LINKED ISSUE #319: Remove `All` in all the methods
It was decided for the consistency between the SDK that the name of the methods retrieving a resource list would be pluralized rather than using the word `All`.
However the followings methods have kept the All in their names:
`GetAllStats`
`GetAllIndexes`
`DeleteAllDocuments`
`GetAllRawIndexes`

It would be nice if they were renamed:
`GetStats`
`GetIndexes`
`DeleteDocuments`
`GetRawIndexes`

### Warning
Not to be implemented before the release of the v0.28.0

### PR DESCRIPTION
**Pull request by vishalsodani**
# Pull Request

## What does this PR do?
Fixes #319 

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -51,11 +51,11 @@ add_or_update_documents_1: |-
   }
   client.Index("movies").UpdateDocuments(documents)
 delete_all_documents_1: |-
-  client.Index("movies").DeleteAllDocuments()
+  client.Index("movies").DeleteDocuments()
```
Comment (alallema) [inline .code-samples.meilisearch.yaml:54]:
```suggestion
  client.Index("movies").DeleteAllDocuments()
```

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -51,11 +51,11 @@ add_or_update_documents_1: |-
...
 delete_one_document_1: |-
   client.Index("movies").DeleteDocument("25684")
 delete_documents_1: |-
-  client.Index("movies").DeleteDocuments([]string{
+  client.Index("movies").DeleteDocumentsInBatch([]string{
```
Comment (alallema) [inline .code-samples.meilisearch.yaml:58]:
```suggestion
  client.Index("movies").DeleteDocuments([]string{
```

File: index.go [hunk 3]
```diff
@@ -35,8 +35,8 @@ type IndexInterface interface {
 	GetDocument(uid string, request *DocumentQuery, documentPtr interface{}) error
 	GetDocuments(param *DocumentsQuery, resp *DocumentsResult) error
 	DeleteDocument(uid string) (resp *TaskInfo, err error)
-	DeleteDocuments(uid []string) (resp *TaskInfo, err error)
-	DeleteAllDocuments() (resp *TaskInfo, err error)
+	DeleteDocumentsInBatch(uid []string) (resp *TaskInfo, err error)
+	DeleteDocuments() (resp *TaskInfo, err error)
```
Comment (alallema) [inline index.go:39]:
```suggestion
	DeleteDocuments(uid []string) (resp *TaskInfo, err error)
	DeleteAllDocuments() (resp *TaskInfo, err error)
```

File: index_documents.go [hunk 4]
```diff
@@ -384,7 +384,7 @@ func (i Index) DeleteDocument(identifier string) (resp *TaskInfo, err error) {
 	return resp, nil
 }
 
-func (i Index) DeleteDocuments(identifier []string) (resp *TaskInfo, err error) {
+func (i Index) DeleteDocumentsInBatch(identifier []string) (resp *TaskInfo, err error) {
```
Comment (alallema) [inline index_documents.go:387]:
```suggestion
func (i Index) DeleteDocuments(identifier []string) (resp *TaskInfo, err error) {
```

File: index_documents.go [hunk 5]
```diff
@@ -393,23 +393,23 @@ func (i Index) DeleteDocuments(identifier []string) (resp *TaskInfo, err error)
 		withRequest:         identifier,
 		withResponse:        resp,
 		acceptedStatusCodes: []int{http.StatusAccepted},
-		functionName:        "DeleteDocuments",
+		functionName:        "DeleteDocumentsInBatch",
```
Comment (alallema) [inline index_documents.go:396]:
```suggestion
		functionName:        "DeleteDocuments",
```

File: index_documents.go [hunk 6]
```diff
@@ -393,23 +393,23 @@ func (i Index) DeleteDocuments(identifier []string) (resp *TaskInfo, err error)
...
 	}
 	if err := i.client.executeRequest(req); err != nil {
 		return nil, err
 	}
 	return resp, nil
 }
 
-func (i Index) DeleteAllDocuments() (resp *TaskInfo, err error) {
+func (i Index) DeleteDocuments() (resp *TaskInfo, err error) {
```
Comment (alallema) [inline index_documents.go:404]:
```suggestion
func (i Index) DeleteAllDocuments() (resp *TaskInfo, err error) {
```

File: index_documents.go [hunk 7]
```diff
@@ -393,23 +393,23 @@ func (i Index) DeleteDocuments(identifier []string) (resp *TaskInfo, err error)
...
 	resp = &TaskInfo{}
 	req := internalRequest{
 		endpoint:            "/indexes/" + i.UID + "/documents",
 		method:              http.MethodDelete,
 		withRequest:         nil,
 		withResponse:        resp,
 		acceptedStatusCodes: []int{http.StatusAccepted},
-		functionName:        "DeleteAllDocuments",
+		functionName:        "DeleteDocuments",
```
Comment (alallema) [inline index_documents.go:412]:
```suggestion
		functionName:        "DeleteAllDocuments",
```

File: index_documents_test.go [hunk 8]
```diff
@@ -952,7 +952,7 @@ func TestIndex_AddDocumentsNdjsonInBatches(t *testing.T) {
 	}
 }
 
-func TestIndex_DeleteAllDocuments(t *testing.T) {
+func TestIndex_DeleteDocuments(t *testing.T) {
```
Comment (alallema) [inline index_documents_test.go:955]:
```suggestion
func TestIndex_DeleteAllDocuments(t *testing.T) {
```

File: index_documents_test.go [hunk 9]
```diff
@@ -963,9 +963,9 @@ func TestIndex_DeleteAllDocuments(t *testing.T) {
 		wantResp *TaskInfo
 	}{
 		{
-			name: "TestIndexBasicDeleteAllDocuments",
+			name: "TestIndexBasicDeleteDocuments",
```
Comment (alallema) [inline index_documents_test.go:966]:
```suggestion
			name: "TestIndexBasicDeleteAllDocuments",
```

File: index_documents_test.go [hunk 10]
```diff
@@ -963,9 +963,9 @@ func TestIndex_DeleteAllDocuments(t *testing.T) {
...
 			args: args{
-				UID:    "TestIndexBasicDeleteAllDocuments",
+				UID:    "TestIndexBasicDeleteDocuments",
```
Comment (alallema) [inline index_documents_test.go:968]:
```suggestion
				UID:    "TestIndexBasicDeleteAllDocuments",
```

File: index_documents_test.go [hunk 11]
```diff
@@ -975,9 +975,9 @@ func TestIndex_DeleteAllDocuments(t *testing.T) {
 			},
 		},
 		{
-			name: "TestIndexDeleteAllDocumentsWithCustomClient",
+			name: "TestIndexDeleteDocumentsWithCustomClient",
```
Comment (alallema) [inline index_documents_test.go:978]:
```suggestion
			name: "TestIndexDeleteAllDocumentsWithCustomClient",
```

File: index_documents_test.go [hunk 12]
```diff
@@ -975,9 +975,9 @@ func TestIndex_DeleteAllDocuments(t *testing.T) {
...
 			args: args{
-				UID:    "TestIndexDeleteAllDocumentsWithCustomClient",
+				UID:    "TestIndexDeleteDocumentsWithCustomClient",
```
Comment (alallema) [inline index_documents_test.go:980]:
```suggestion
				UID:    "TestIndexDeleteAllDocumentsWithCustomClient",
```

File: index_documents_test.go [hunk 13]
```diff
@@ -994,7 +994,7 @@ func TestIndex_DeleteAllDocuments(t *testing.T) {
 			t.Cleanup(cleanup(c))
 
 			SetUpBasicIndex(tt.args.UID)
-			gotResp, err := i.DeleteAllDocuments()
+			gotResp, err := i.DeleteDocuments()
```
Comment (alallema) [inline index_documents_test.go:997]:
```suggestion
			gotResp, err := i.DeleteAllDocuments()
```

File: index_documents_test.go [hunk 14]
```diff
@@ -1154,7 +1154,7 @@ func TestIndex_DeleteOneDocument(t *testing.T) {
 	}
 }
 
-func TestIndex_DeleteDocuments(t *testing.T) {
+func TestIndex_DeleteDocumentsInBatch(t *testing.T) {
```
Comment (alallema) [inline index_documents_test.go:1157]:
```suggestion
func TestIndex_DeleteDocuments(t *testing.T) {
```

File: index_documents_test.go [hunk 15]
```diff
@@ -1167,7 +1167,7 @@ func TestIndex_DeleteDocuments(t *testing.T) {
 		wantResp *TaskInfo
 	}{
 		{
-			name: "TestIndexBasicDeleteDocument",
+			name: "TestIndexBasicDeleteDocumentInBatch",
```
Comment (alallema) [inline index_documents_test.go:1170]:
```suggestion
			name: "TestIndexBasicDeleteDocument",
```

File: index_documents_test.go [hunk 16]
```diff
@@ -1183,7 +1183,7 @@ func TestIndex_DeleteDocuments(t *testing.T) {
 			},
 		},
 		{
-			name: "TestIndexDeleteDocumentWithCustomClient",
+			name: "TestIndexDeleteDocumentInBatchWithCustomClient",
```
Comment (alallema) [inline index_documents_test.go:1186]:
```suggestion
			name: "TestIndexDeleteDocumentWithCustomClient",
```

File: index_documents_test.go [hunk 17]
```diff
@@ -1199,7 +1199,7 @@ func TestIndex_DeleteDocuments(t *testing.T) {
 			},
 		},
 		{
-			name: "TestIndexBasicDeleteDocument",
+			name: "TestIndexBasicDeleteDocumentInBatch",
```
Comment (alallema) [inline index_documents_test.go:1202]:
```suggestion
			name: "TestIndexBasicDeleteDocument",
```

File: index_documents_test.go [hunk 18]
```diff
@@ -1217,7 +1217,7 @@ func TestIndex_DeleteDocuments(t *testing.T) {
...
```
Comment (alallema) [inline index_documents_test.go:1220]:
```suggestion
			name: "TestIndexBasicDeleteDocument",
```

File: index_documents_test.go [hunk 19]
```diff
@@ -1246,7 +1246,7 @@ func TestIndex_DeleteDocuments(t *testing.T) {
 
 			testWaitForTask(t, i, gotAddResp)
 
-			gotResp, err := i.DeleteDocuments(tt.args.identifier)
+			gotResp, err := i.DeleteDocumentsInBatch(tt.args.identifier)
```
Comment (alallema) [inline index_documents_test.go:1249]:
```suggestion
			gotResp, err := i.DeleteDocuments(tt.args.identifier)
```

**Review by alallema — COMMENTED** [review]
Thanks, @vishalsodani for this PR 🎉 
I made a mistake in the description of the PR I'm so sorry 😓 ... `DeleteAllDocuments` and `DeleteDocuments` doesn't have to be changed I made all the suggestions to fix it but you can fix it by yourself if you prefer 😊 😅  
Thanks again for contributing with Meilisearch :heart:
If you are participating in [Hacktoberfest](https://blog.meilisearch.com/contribute-hacktoberfest-2022/), and you would like to receive a small gift from Meilisearch too, don't forget to fill [this form](https://meilisearch.typeform.com/to/idt5qugY).

**Comment by vishalsodani** [comment]
> Thanks, @vishalsodani for this PR tada I made a mistake in the description of the PR I'm so sorry sweat ... `DeleteAllDocuments` and `DeleteDocuments` doesn't have to be changed I made all the suggestions to fix it but you can fix it by yourself if you prefer blush sweat_smile Thanks again for contributing with Meilisearch heart If you are participating in [Hacktoberfest](https://blog.meilisearch.com/contribute-hacktoberfest-2022/), and you would like to receive a small gift from Meilisearch too, don't forget to fill [this form](https://meilisearch.typeform.com/to/idt5qugY).

Thanks for your review @alallema. Please go ahead and merge all the changes you have done.

**Review by alallema — APPROVED** [review]
Thanks, @vishalsodani for this PR!
LGTM 🎉

**Comment by alallema** [comment]
bors merge

**Comment by alallema** [comment]
Thanks again for contributing with Meilisearch :heart:
If you are participating in [Hacktoberfest](https://blog.meilisearch.com/contribute-hacktoberfest-2022/), and you would like to receive a small gift from Meilisearch too, don't forget to fill [this form](https://meilisearch.typeform.com/to/idt5qugY).

**Pull request merged**